arch/risc-v/src/common/riscv_swint.c

```diff
@@ -455,7 +455,7 @@ int riscv_swint(int irq, void *context, void *arg)
 
           rtcb->flags         |= TCB_FLAG_SYSCALL;
 #else
-          svcerr("ERROR: Bad SYS call: %" PRId64 "\n", regs[REG_A0]);
+          svcerr("ERROR: Bad SYS call: %" PRIdPTR "\n", regs[REG_A0]);
```

Comment:
Maybe can switch to `PRIxREG` here.
Also rework `static void riscv_registerdump(const uint64_t *regs)` to use `uintptr_t`.